enrollment-server/src/main/java/com/wultra/app/enrollmentserver/impl/service/ActivationCodeDelegate.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * PowerAuth Enrollment Server
+ * Copyright (C) 2021 Wultra s.r.o.
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package com.wultra.app.enrollmentserver.impl.service;
+
+import java.util.List;
+
+/**
+ * Callback provider for lifecycle events of activation code.
+ *
+ * @author Petr Dvorak, petr@wultra.com
+ */
+public interface ActivationCodeDelegate {
+
+    /**
+     * Fetch destination application ID value based in application ID. Check if the source app can
+     * activate the destination one.
+     *
+     * @param applicationId Application identifier for app lookup.
+     * @param sourceAppId Source application ID.
+     * @param activationFlags Activation flags.
+     * @param applicationRoles Application roles.
+     * @return Destination application ID.
+     */
+    Long destinationApplicationId(String applicationId, Long sourceAppId, List<String> activationFlags, List<String> applicationRoles);
```

Comment:
The method name is missing a verb. The other methods have a verb (`add`, `did`), so the method name seems a bit inconsistent.

Reply:
I will add "fetch"